ompi/attribute/attribute.c

```diff
@@ -687,13 +742,44 @@ int ompi_attr_set_c(ompi_attribute_type_t type, void *object,
 
 
 /*
+ * Front-end function internally called by the C API functions to set an
+ * int attribute.
```

Comment:
DISP_UNIT is an int * in C.
we used to treat it internally as an Fint, which is incorrect with -i8
since it is read only, there is no need for a pointer, and storing the value makes things much easier.